app/src/main/java/com/halim/adam/cashmaster/AddNewSpending.java

```diff
@@ -33,9 +42,11 @@ protected void onCreate(Bundle savedInstanceState) {
         ArrayList<String> categoryStrings = new ArrayList<>();
         ArrayList<String> budgetStrings = new ArrayList<>();
         for(int c = 0; c < categoryList.size(); c++){
+            categoryIdList.add(categoryList.get(c).getId());
             categoryStrings.add(categoryList.get(c).getName());
         }
         for(int c = 0; c < budgetRatioList.size(); c++){
+            budgetIdList.add(budgetRatioList.get(c).getId());
             budgetStrings.add(budgetRatioList.get(c).getName());
         }
```

Comment:
The item obtained from the list (categoryList.get(c) and budgetRatioList.get(c) can be saved into a variable before getting the id and name so the arraylist will only be accessed once